src/ring_theory/noetherian.lean

```diff
@@ -69,6 +70,20 @@ theorem fg_def {N : submodule R M} :
   exact ⟨t, rfl⟩
 end⟩
 
+lemma fg_iff_add_monoid_fg :
+  (⊤ : submodule ℕ M).fg ↔ add_monoid.fg M :=
```

Comment:
Oh, this should probably be stated as:
```suggestion
lemma fg_iff_add_submonoid_fg (S : submodule ℕ M) :
  S.fg ↔ S.to_add_submonoid.fg :=
```

Reply:
And then the lemma as you have it here would be:
```suggestion
lemma finite_iff_add_monoid_fg :
  module.finite ℕ M ↔ add_monoid.fg M :=
```
and would move to the file that defines `module.finite`

Reply:
`module.finite` is defined only for module over a `ring`... let me see what happens if I generalize it to `semiring` (just the definition, without touching any lemma).

Reply:
Ah; in that case, perhaps just make the `fg_iff_add_submonoid_fg` change in this PR, and we can deal with `module.finite` later?